tests/lib/fakeinstaller/mk-classic-rootfs.sh

```diff
@@ -0,0 +1,87 @@
+#!/bin/bash
+
+set -e
+
+# uncomment for better debug messages
+#set -x
+#exec > /tmp/mk-classic-rootfs.sh.log
+#exec 2>&1
+
+
+# XXX: merge with the work from alfonso in
+# tests/nested/manual/fde-on-classic/mk-image.sh (PR:12102)
+create_classic_rootfs() {
+    set -x
+    local DESTDIR="$1"
+    local BASETAR="$2"
+
+    # Copy base filesystem
+    sudo tar -C "$DESTDIR" -xf "$BASETAR"
+
+    # Create basic devices to be able to install packages
+    [ -e "$DESTDIR"/dev/null ] || sudo mknod -m 666 "$DESTDIR"/dev/null c 1 3
+    [ -e "$DESTDIR"/dev/zero ] || sudo mknod -m 666 "$DESTDIR"/dev/zero c 1 5
+    [ -e "$DESTDIR"/dev/random ] || sudo mknod -m 666 "$DESTDIR"/dev/random c 1 8
+    [ -e "$DESTDIR"/dev/urandom ] || sudo mknod -m 666 "$DESTDIR"/dev/urandom c 1 9
+    # ensure resolving works inside the chroot
+    echo "nameserver 8.8.8.8" | sudo tee -a "$DESTDIR"/etc/resolv.conf
+    # install additional packages
+    sudo chroot "$DESTDIR" /usr/bin/sh -c "DEBIAN_FRONTEND=noninteractive apt update"
+    local pkgs="snapd ssh openssh-server sudo iproute2 iputils-ping isc-dhcp-client netplan.io vim-tiny kmod cloud-init"
+    sudo chroot "$DESTDIR" /usr/bin/sh -c \
+         "DEBIAN_FRONTEND=noninteractive apt install --no-install-recommends -y $pkgs"
+    # netplan config
+    cat <<'EOF' | sudo tee "$DESTDIR"/etc/netplan/00-ethernet.yaml
+network:
+  ethernets:
+    any:
+      match:
+        name: e*
+      dhcp4: true
+  version: 2
+EOF
+
+    # ensure we can login
+    sudo chroot "$DESTDIR" /usr/sbin/adduser --disabled-password --gecos "" user1
+    printf "ubuntu\nubuntu\n" | sudo chroot "$DESTDIR" /usr/bin/passwd user1
+    echo "user1 ALL=(ALL) NOPASSWD:ALL" | sudo tee -a "$DESTDIR"/etc/sudoers
+
+    # set password for root user
+    sudo chroot "$DESTDIR" /usr/bin/sh -c 'echo root:root | chpasswd'
+    sudo tee -a "$DESTDIR/etc/ssh/sshd_config" <<'EOF'
+PermitRootLogin yes
+PasswordAuthentication yes
+EOF
+
+    # install the current in-development version of snapd when available,
+    # this will give us seeding support
+    GOPATH="${GOPATH:-./}"
+    package=$(find "$GOPATH" -maxdepth 1 -name "snapd_*.deb")
+    if [ -e "$package"  ]; then
+        cp "$package" "$DESTDIR"/var/cache/apt/archives
+        sudo chroot "$DESTDIR" /usr/bin/sh -c \
+             "DEBIAN_FRONTEND=noninteractive apt install -y /var/cache/apt/archives/$(basename "$package")"
+    fi
+
+    # ensure that we have a mount point for the bind mount below
+    sudo mkdir -p "$DESTDIR"/boot/grub
+    # TODO: find out what is doing this on UC20+ systems
```

Comment:
MIght be worthwhile to update this comment. One core we do this via https://github.com/snapcore/core-initrd/blob/main/factory/usr/lib/the-modeenv#L13 and `the-modeenv` is run via: https://github.com/snapcore/core-initrd/blob/main/factory/usr/lib/systemd/system/populate-writable.service#L25 - we don't run the `populate-writable` service on classic so this is not being done. Probably fine to keep this workaround for now (but the comment update might be nice)

Reply:
Updated now